spec/integration/util/windows/user_spec.rb

```diff
@@ -155,6 +156,20 @@ def expect_logon_failure_error(&block)
       end
     end
 
+    describe 'logon_user_by_logon_type' do
+      let(:password){'test'}
+      let(:fLOGON32_LOGON_INTERACTIVE){2}
+      let(:fLOGON32_LOGON_NETWORK){3}
+      let(:fLOGON32_PROVIDER_DEFAULT){0}
+      let(:token){'test_token'}
+      it 'should raise error given that logon returns false' do
+        allow(Puppet::Util::Windows::User.logon_user_by_logon_type).to receive(:LogonUserW).and_return(false)
+        expect(Puppet::Util::Windows::User.logon_user_by_logon_type(:user, :password, :fLOGON32_LOGON_INTERACTIVE,fLOGON32_PROVIDER_DEFAULT, token))
```

Comment:
I don't see where the `:user` symbol is ever defined. I also don't understand the inconsistency in the way the parameters are called between referring to them by symbols (they are all defined as symbols in the `let` calls) vs calling them by regular variable names.

Reply:
The short answer for why `let(:foo)` allows you do do`foo` later on in the tests is that `let` defines a function with the same name as its argument.

https://relishapp.com/rspec/rspec-core/docs/helper-methods/let-and-let

The `:user` symbol doesn't need to be defined anywhere. You can think of symbols as special kinds of string literals in Ruby.

Reply:
Right, I get those parts. But the question remains about the inconsistency with which they are called on this line. For example with `:fLOGON32_LOGON_INTERACTIVE`, is that returning the defined value of 2, or since it's a symbol is just a symbol? I see `fLOGON32_PROVIDER_DEFAULT` and `token` are called as functions which makes sense, but are both the symbolized methods of calling and the function method of calling both valid? If they are, why choose one over the other, and especially why both?

Reply:
`:fLOGON32_LOGON_INTERACTIVE` is just a symbol. It doesn't have a separate value other than just being a symbol.

Reply:
The `let()` on 161 defines it a s`2`.

Reply:
As for why one is a symbol, and why the other one isn't, I don't know off hand. I haven't dug into the Windows API that's being used here.

Reply:
So it looks like with this call, the parameters should be regular references to the functions provided by the `let()` definitions above. A symbolized parameter value to this method seems likely to break the method in a way that satisfies the test but does not test what it thinks it's testing, and results in an invalid test.

Also please define a `let()` for the value of the `user` parameter, as that will need to be passed in as a real value as well.